src/main/java/org/candlepin/subscriptions/security/IdentityHeaderAuthenticationDetailsSource.java

```diff
@@ -56,60 +52,56 @@
     private static final Logger log =
         LoggerFactory.getLogger(IdentityHeaderAuthenticationDetailsSource.class);
 
-    // NB: Right now there are just two roles.  If we expand on that, we should externalize all the
-    //  roles to an enum, serialize the header JSON to an object, and write an actual predicate method
-    //  isUserInRole
-    public static final String ORG_ADMIN_ROLE = "ORG_ADMIN";
-    public static final String INTERNAL_ROLE = "INTERNAL";
-
-    private ObjectMapper objectMapper;
     private Attributes2GrantedAuthoritiesMapper authMapper;
     private ApplicationProperties props;
+    private RbacApi rbac;
+    private String rbacApplicationName;
+    private RoleProvider roleProvider;
 
-    public IdentityHeaderAuthenticationDetailsSource(ApplicationProperties props, ObjectMapper objectMapper,
-        Attributes2GrantedAuthoritiesMapper authMapper) {
-        this.objectMapper = objectMapper;
+    public IdentityHeaderAuthenticationDetailsSource(ApplicationProperties props,
+        Attributes2GrantedAuthoritiesMapper authMapper,
+        RbacApi rbac, String rbacApplicationName) {
         this.authMapper = authMapper;
         this.props = props;
+        this.rbac = rbac;
+        this.rbacApplicationName = rbacApplicationName;
 
         if (props.isDevMode()) {
             log.info("Running in DEV mode. Security will be disabled.");
         }
+        this.roleProvider = new RoleProvider(rbacApplicationName, props.isDevMode());
     }
 
-    protected Collection<String> getUserRoles(HttpServletRequest context) {
-        ArrayList<String> userRolesList = new ArrayList<>();
-        try {
-            String identityHeader = context.getHeader(RH_IDENTITY_HEADER);
-
-            if (StringUtils.isEmpty(identityHeader)) {
-                return userRolesList;
-            }
-
-            byte[] decodedHeader = Base64.getDecoder().decode(identityHeader);
-            Map authObject = objectMapper.readValue(decodedHeader, Map.class);
-            Map identity = (Map) authObject.getOrDefault("identity", Collections.emptyMap());
-            Map user = (Map) identity.getOrDefault("user", Collections.emptyMap());
-
-            if (Boolean.TRUE.equals(user.get("is_org_admin")) || props.isDevMode()) {
-                userRolesList.add(ORG_ADMIN_ROLE);
+    protected Collection<String> getUserRoles() {
+        List<String> permissions = new LinkedList<>();
+        if (!props.isDevMode()) {
+            try {
+                // Get all permissions for the configured application name.
+                List<Access> accessList = rbac.getCurrentUserAccess(rbacApplicationName);
+                for (Access access : accessList) {
+                    if (access.getPermission() != null) {
+                        permissions.add(access.getPermission());
+                    }
+                }
```

Comment:
```suggestion
permissions = accessList.stream().map(Access::getPermission).filter(Objects::nonNull).collect(Collectors.toCollection(LinkedList::new));
```

It's even shorter if you just use `Collectors.toList()`.  Are we sure we want to use a `LinkedList` here?  The memory consumption is higher due to maintaining all the pointers, and if we allocate the ArrayList at accessList.size(), we'll avoid the hit of having to resize the list.

Reply:
DONE - Addressed in next commit.